python/src/dataset.rs

```diff
@@ -509,22 +508,23 @@ impl Dataset {
 pub fn write_dataset(reader: &PyAny, uri: &str, options: &PyDict) -> PyResult<bool> {
     let params = get_write_params(options)?;
     Runtime::new()?.block_on(async move {
-        let mut batches: Box<dyn RecordBatchReader> = if reader.is_instance_of::<Scanner>()? {
+        if reader.is_instance_of::<Scanner>() {
             let scanner: Scanner = reader.extract()?;
-            Box::new(
-                scanner
-                    .to_reader()
-                    .await
-                    .map_err(|err| PyValueError::new_err(err.to_string()))?,
-            )
+            let batches = scanner
```

Comment:
Lets do 

```
let batches = if reader.is_istance_of::<Scanner>() {
   ...
} else {
   ...
}

LanceDataset::write(batches, uri, params)...

```

Reply:
The only reason I split this is because we can't pass a Box anymore, and the two arms of the if-else statement have different RBR types, so they need to be boxed if they are to go to the same variable. So for now we can't do this, unless we wait until the Boxed RBR is implemented upstream.

Reply:
Got it. It is fine. let's merge.